Bottom: c6b8acb4c91fe4fb0fafca5cfd73b71b3b474710
Top:    1173e5d2663168e64dbf28bed9a753a5ae060e71
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-09-05 15:11:14 +0800

vhost: batch signalling

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index b304659..be7472a 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -590,7 +590,6 @@ static void handle_tx(struct vhost_net *net)
 					" len %d != %zd\n", err, len);
 			if (!zcopy) {
 				vhost_add_used_idx(vq, 1);
-				vhost_signal(&net->dev, vq);
 			} else if (!zcopy_used) {
 				vhost_add_used_and_signal(&net->dev,
 							  vq, head, 0);
@@ -602,6 +601,8 @@ static void handle_tx(struct vhost_net *net)
 				goto out;
 			}
 		}
+		if (!zcopy)
+			vhost_signal(&net->dev, vq);
 	}
 out:
 	mutex_unlock(&vq->mutex);
